guides/updates/administration/vue3.md

```diff
@@ -0,0 +1,129 @@
+You can participate in discussions and seek help on the Shopware community Slack. There is a dedicated channel for this topic called `#vue3-update`.
+
+## External resources
+
+Here is a handpicked selection of external resources. This list provides a handy reference, granting you access to all the essential materials you might need.
+
+- (Vue 3 migration guide)[https://v3-migration.vuejs.org]
+- (Vue 3 breaking changes)[https://v3-migration.vuejs.org/breaking-changes/]
+- (Vue router migration guide)[https://router.vuejs.org/guide/migration/]
+- (Vue test utils migration guide)[https://test-utils.vuejs.org/migration/]
+
+## Step-by-step guide
+::: warning
+Depending on when you read this document, you still need to enable the `VUE3` feature flag.
+To be safe, enable the `VUE3` feature flag and restart your watcher.
+:::
+
+To follow along, you should have the following:
+
+- the latest Shopware `trunk` or an official release candidate
+- installed and activated your plugin
+- a running administration watcher (`composer run watch:admin`)
+
+### Update your plugin npm dependencies
+
+Make sure to align your `package.json` dependencies with the administration.
+
+### Verify the Admin is using Vue 3
+
+If you evaluate the following expression in the browser dev tools, the value `true` must be returned.
+
+```javascript
+Shopware.Application.view.vue3
+```
+
+### Check your templates
+
+For your templates to work correctly, follow the following in no specific order:
+
+- Replace all `sw-field` usages with the corresponding components.
+- (Check all v-models)[https://v3-migration.vuejs.org/breaking-changes/v-model.html]
+- (Check for deprecated slot syntax)[https://eslint.vuejs.org/rules/no-deprecated-slot-attribute.html]
+- (Check router-view transition combinations)[https://router.vuejs.org/guide/migration/#-router-view-keep-alive-and-transition-]
+- (Check your key attributes)[https://v3-migration.vuejs.org/breaking-changes/key-attribute.html]
+
+### Check your code
+
+Most of your code should be unaffected by the update. You can start by searching for `this.$`. The usage of `this.$` is an indicator of Vue's internal API. These calls are very likely to break except for `this.$tc`.
+
+If you have a lot of Vue internal API calls, check out the (Known issues section)[#known-issues].
+The best way to find errors is to test your application thoroughly, either by hand or automated.
+
+## Known issues
+
+### Lifecycle hooks
+Lifecycle hooks such as `@hook:mounted` may be triggered multiple times if the component is loaded asynchronously. Vue 3 will emit the hook for the `AsyncComponentWrapper` and the underlying component. Expect lifecycle hooks to be triggered more than once.
+
+### Using slots programmatically
+It is no longer sufficient to check if `this.$slots` has a property with the slot name to see if that slot exists. Instead, you must verify if your `slotName` contains an actual `v-node`.
```

Comment:
I'm sure I used `$slots.mySlot` in a vue3 template the other day to check if a slot was provided, is that something different?

Reply:
In Vue 3 $slots.MySlot should always be a render function. Accroding to the official docs you have to call this render function and see if it returns a v-node.

Reply:
Small addition: check for the slot content is quite complex, depending on what you define as "empty" slot:
Discussion: https://github.com/vuejs/core/issues/4733#issuecomment-1244011992
WIP implementation (in vitepress): https://github.com/vuejs/vitepress/pull/2297/files#diff-eed5fa858ff4a0d3291cfeaa80f5426ed1948b75f25f9fafc002efe82a91bc36

Reply:
Thank you for sharing :+1: Pretty interesting discussion :blue_heart: